app/scripts/controllers/membership.js

```diff
@@ -0,0 +1,228 @@
+          name: ''
+        },
+        mode: {
+          edit: false
+        },
+        selectTab: function(selected) {
+          $scope.newBinding.kind = selected;
+        }
+      });
+
+      DataService.list('projects', {}, function(resp) {
+        angular.extend($scope, {
+          projects: _.map(resp.by('metadata.name'), function(project) {
+            return project.metadata.name;
+          })
+        });
+      });
+
+      var createModalScope = function(subjectName, kind, role, currentUserName) {
+        var modalScope = {
+          alerts: {},
+          message: 'Are you sure you want to remove "' + role + '" from the ' + formatKind(kind) + ' "' + subjectName + '"?',
+          okButtonText: 'Remove',
+          okButtonClass: 'btn-danger',
+          cancelButtonText: 'Cancel'
+        };
+        if(_.isEqual(subjectName, currentUserName)) {
+          modalScope.message = 'Are you sure you want to remove "' + role + '" from ' + subjectName + ' (yourself)? ';
+          showAlert('currentUserLastRole', 'error', 'This is your last role in the project!', null, modalScope);
+        }
+        if(_.isEqual(kind, 'ServiceAccount')) {
+          showAlert('editingServiceAccountRole', 'error', 'Removing roles granted to service accounts may break the system.', null, modalScope);
+        }
+        return modalScope;
+      };
+
+      ProjectsService
+        .get($routeParams.project)
+        .then(_.spread(function(project, context) {
+          requestContext = context;
+          refreshRoleBindingList();
+          angular.extend($scope, {
+            project: project,
+            subjectKinds: subjectKinds,
+            confirmRemove: function(subjectName, kind, role) {
+              var redirectToProjectList = null;
+              var modalScope = createModalScope(subjectName, kind, role, user.metadata.name);
+              if(_.isEqual(subjectName, user.metadata.name)) {
+                if(MembershipService.isLastRole(user.metadata.name, $scope.roleBindings)) {
+                  redirectToProjectList = true;
+                }
+              }
+              $uibModal.open({
+                animation: true,
+                templateUrl: 'views/modals/confirm.html',
+                controller: 'ConfirmModalController',
+                resolve: {
+                  modalConfig: function() {
+                    return modalScope;
+                  }
+                }
+              }).result.then(function() {
+                RoleBindingsService
+                  .removeSubject(subjectName, role, $scope.roleBindings, requestContext)
+                  .then(function() {
+                    if(redirectToProjectList) {
+                      $location.url("./");
+                    } else {
+                      refreshRoleBindingList();
+                      showAlert('rolebindingUpdate', 'success', 'The role ' + role + ' was removed from ' + subjectName);
```

Comment:
and yes i realize the irony that i told you to use punctuation and then didn't use punctuation in the process ;-)